go/vt/vtgate/planbuilder/physical/route_planning.go

```diff
@@ -226,6 +226,19 @@ func createPhysicalOperatorFromDelete(ctx *plancontext.PlanningContext, op *abst
 		return nil, err
 	}
 
+	if opCode == engine.Unsharded {
+		return &Route{
+			Source: &Delete{
+				QTable: op.Table,
+				VTable: vindexTable,
+				AST:    op.AST,
+			},
+			RouteOpCode:       opCode,
+			Keyspace:          vindexTable.Keyspace,
+			TargetDestination: dest,
+		}, nil
+	}
+
 	primaryVindex, vindexAndPredicates, err := getVindexInformation(op.TableID(), op.Table.Predicates, vindexTable)
 	if err != nil {
```

Comment:
instead of this let's do the same things we do in `Update` in `getVindexInformation`  check for the table keyspace if unsharded, return immediately. 

This is clearly a miss(BUG) and will impact any delete query for an unsharded table if that table is declared in vschema.